deps/build.jl

```diff
@@ -35,9 +35,6 @@ else
                     if libhdf5_size != filesize(Libdl.dlpath(libhdf5))
                         error("HDF5 library has changed, re-run Pkg.build(\\\"HDF5\\\")")
                     end
-                    if libversion < v"1.12"
-                        error("HDF5.jl requires ≥ v1.12 of the HDF5 library.")
```

Comment:
```
@warn  "HDF5.jl requires ... use at your own risk ...
```

Reply:
Sounds good.

Maybe it would be better if it's only printed when HDF5.jl is built (here it would be printed every time HDF5.jl is loaded), but I don't know if that's possible.

Or maybe we could read an environment variable which would disable warnings, as in:
```julia
if libversion < v"1.12" && get(ENV, "JULIA_HDF5_NOWARN", "0") == "0"
    @warn "HDF5.jl requires ≥ v1.12 of the HDF5 library, but v\$libversion was detected. Use at your own risk..."
end
```

Reply:
Probably better to check if `JULIA_HDF5_LIBRARY_PATH ` is not set to `nothing`

Reply:
Or actually move this warning up in the file instead. (i.e. dont' write this out? )

Reply:
I just tried this, but the problem is that by default nothing is printed to the terminal when building a package. This is unless `--verbose` is passed.